lib/mayaHydra/mayaPlugin/plugin.cpp

```diff
@@ -70,14 +70,13 @@ namespace {
 
     std::vector<MCallbackId> _pluginLoadingCallbackIds;
 
-    void setEnvVar(const char* envVarSet)
+    void setEnv(const std::string& name, const std::string& value)
     {
-        // putenv requires char* and I'm not willing to use const cast!
-        const auto        envVarSize = strlen(envVarSet) + 1;
-        std::vector<char> envVarData;
-        envVarData.resize(envVarSize);
-        snprintf(envVarData.data(), envVarSize, "%s", envVarSet);
-        putenv(envVarData.data());
```

Comment:
As per
https://stackoverflow.com/questions/5873029/questions-about-putenv-and-setenv
putenv does not make a copy of the argument string.  Therefore, in this code, as soon as the function exits and the memory allocated for envVarData is re-used, the environment will be changed / corrupted.  setenv() is the proper choice.